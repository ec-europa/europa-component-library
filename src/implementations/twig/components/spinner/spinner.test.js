import {
  merge,
  renderTwigFileAsNode,
  renderTwigFileAsHtml,
} from '@ecl/test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';

import dataDefault from '@ecl/specs-component-spinner/demo/data';

expect.extend(toHaveNoViolations);

describe('Spinner', () => {
  const template = '@ecl/spinner/spinner.html.twig';
  const render = (params) => renderTwigFileAsNode(template, params);

  describe('ECL Spinner', () => {
    test('primary renders correctly', () => {
      expect.assertions(1);

      return expect(render(dataDefault)).resolves.toMatchSnapshot();
    });

    test('negative renders correctly', () => {
      expect.assertions(1);

      return expect(
        render({ ...dataDefault, variant: 'negative' })
      ).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra class names', () => {
      expect.assertions(1);

      const withExtraClasses = merge(dataDefault, {
        extra_classes: 'custom-table--test',
      });

      return expect(render(withExtraClasses)).resolves.toMatchSnapshot();
    });

    test('renders correctly with extra attributes', () => {
      expect.assertions(1);

      const withExtraAttributes = merge(dataDefault, {
        extra_attributes: [
          { name: 'data-test', value: 'data-test-value' },
          { name: 'data-test-1', value: 'data-test-value-1' },
        ],
      });

      return expect(render(withExtraAttributes)).resolves.toMatchSnapshot();
    });

    test(`passes the accessibility tests`, async () => {
      expect(
        await axe(await renderTwigFileAsHtml(template, dataDefault, true))
      ).toHaveNoViolations();
    });
  });
});
