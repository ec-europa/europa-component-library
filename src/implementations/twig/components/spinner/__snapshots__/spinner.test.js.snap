// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner ECL Spinner negative renders correctly 1`] = `
<jest>
  <div
    class="ecl-spinner ecl-spinner--negative ecl-spinner--medium ecl-spinner--centered ecl-spinner--visible"
  >
    <svg
      class="ecl-spinner__loader"
      viewBox="25 25 50 50"
    >
      <circle
        class="ecl-spinner__circle"
        cx="50"
        cy="50"
        fill="none"
        r="20"
        stroke-miterlimit="10"
        stroke-width="4px"
        vector-effect="non-scaling-stroke"
      />
    </svg>
    <div
      class="ecl-spinner__text"
    >
      Loading...
    </div>
  </div>
</jest>
`;

exports[`Spinner ECL Spinner primary renders correctly 1`] = `
<jest>
  <div
    class="ecl-spinner ecl-spinner--primary ecl-spinner--medium ecl-spinner--centered ecl-spinner--visible"
  >
    <svg
      class="ecl-spinner__loader"
      viewBox="25 25 50 50"
    >
      <circle
        class="ecl-spinner__circle"
        cx="50"
        cy="50"
        fill="none"
        r="20"
        stroke-miterlimit="10"
        stroke-width="4px"
        vector-effect="non-scaling-stroke"
      />
    </svg>
    <div
      class="ecl-spinner__text"
    >
      Loading...
    </div>
  </div>
</jest>
`;

exports[`Spinner ECL Spinner renders correctly with extra attributes 1`] = `
<jest>
  <div
    class="ecl-spinner ecl-spinner--primary ecl-spinner--medium ecl-spinner--centered ecl-spinner--visible"
    data-test="data-test-value"
    data-test-1="data-test-value-1"
  >
    <svg
      class="ecl-spinner__loader"
      viewBox="25 25 50 50"
    >
      <circle
        class="ecl-spinner__circle"
        cx="50"
        cy="50"
        fill="none"
        r="20"
        stroke-miterlimit="10"
        stroke-width="4px"
        vector-effect="non-scaling-stroke"
      />
    </svg>
    <div
      class="ecl-spinner__text"
    >
      Loading...
    </div>
  </div>
</jest>
`;

exports[`Spinner ECL Spinner renders correctly with extra class names 1`] = `
<jest>
  <div
    class="ecl-spinner ecl-spinner--primary ecl-spinner--medium ecl-spinner--centered ecl-spinner--visible"
  >
    <svg
      class="ecl-spinner__loader"
      viewBox="25 25 50 50"
    >
      <circle
        class="ecl-spinner__circle"
        cx="50"
        cy="50"
        fill="none"
        r="20"
        stroke-miterlimit="10"
        stroke-width="4px"
        vector-effect="non-scaling-stroke"
      />
    </svg>
    <div
      class="ecl-spinner__text"
    >
      Loading...
    </div>
  </div>
</jest>
`;
